src/components/general/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import ErrorMessage, { ErrorMessageProps, ErrorTypes } from "../ErrorMessage";
+
+export default class ErrorBoundary extends React.Component<ErrorMessageProps> {
+    static defaultProps = {
+        errorType: ErrorTypes.error,
+    };
+
+    state = {
+        hasError: false,
+        error: null,
+        errorInfo: null,
+        errorMessage: "",
+    };
+
+    componentDidCatch(error, errorInfo) {
+        this.setState({
+            hasError: true,
+            error,
+            errorInfo,
+            errorMessage: error.message || ""
+        });
+    }
+
+    getErrorMessage() {
+        const { errorMessage } = this.state;
+        const { message } = this.props;
+
+        if (message) {
+            return message;
+        }
+        if (errorMessage !== "") {
+            return errorMessage;
+        }
+
+        return "Unhandled error message";
+    }
+
+    renderChildren() {
+        if (this.state.hasError) {
+            return null;
+        }
+
+        return this.props.children;
+    }
+
+    onButtonClick() {
```

Comment:
Sould this be something more along the lines of "takeAction"? E.g takeAction = () => { if(this.state.hasError && this.props.onTakeAction) { return this.props.onTakeAction(); } window.location.reload(); }

The same would apply to the ErrorMessage component. Makes more sense to "Take action" when something goes wrong than to "button" 🙈

Reply:
Agreed, rewrote method and changed prop name from onButtonClick to onTakeAction